pkg/agent/sysadvisor/metacache/metacache.go

```diff
@@ -77,6 +78,8 @@ type MetaReader interface {
 	// GetPodsInferenceResult gets specified model inference result
 	GetInferenceResult(modelName string) (interface{}, error)
 
+	GetHeadroom(name types.QoSResourceName) (*resource.Quantity, error)
```

Comment:
Not recommended to add GetHeadroom interface because it is not general. Use GetNodeInfo instead and update headroom property.